src/EFCore.PG.NTS/Query/ExpressionTranslators/Internal/NpgsqlNetTopologySuiteMethodCallTranslatorPlugin.cs

```diff
@@ -28,94 +31,91 @@ public class NpgsqlNetTopologySuiteMethodCallTranslatorPlugin : IMethodCallTrans
     public class NpgsqlGeometryMethodTranslator : IMethodCallTranslator
     {
         static readonly MethodInfo _collectionItem =
-            typeof(IGeometryCollection).GetRuntimeProperty("Item").GetMethod;
+            typeof(GeometryCollection).GetRuntimeProperty("Item").GetMethod;
+
+        readonly ISqlExpressionFactory _sqlExpressionFactory;
+        readonly IRelationalTypeMappingSource _typeMappingSource;
+
+        public NpgsqlGeometryMethodTranslator(ISqlExpressionFactory sqlExpressionFactory, IRelationalTypeMappingSource typeMappingSource)
+        {
+            _sqlExpressionFactory = sqlExpressionFactory;
+            _typeMappingSource = typeMappingSource;
+        }
 
         /// <inheritdoc />
         [CanBeNull]
-        public virtual Expression Translate(MethodCallExpression e, IDiagnosticsLogger<DbLoggerCategory.Query> logger)
+        public virtual SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadOnlyList<SqlExpression> arguments)
         {
-            if (!typeof(IGeometry).IsAssignableFrom(e.Method.DeclaringType))
+            if (!typeof(Geometry).IsAssignableFrom(method.DeclaringType))
                 return null;
 
-            switch (e.Method.Name)
+            var geometryExpressions = new[] { instance }.Concat(
+                arguments.Where(e => typeof(Geometry).IsAssignableFrom(e.Type)));
+            var typeMapping = ExpressionExtensions.InferTypeMapping(geometryExpressions.ToArray());
```

Comment:
`geometryExpressions` is used only once, makes sense to inline it. By the way, there is the `Prepend` method.

```csharp
var typeMapping = ExpressionExtensions.InferTypeMapping(
    arguments.Prepend(instance).Where(e => typeof(Geometry).IsAssignableFrom(e.Type)).ToArray());
```

Reply:
Copy-pasted from upstream, but I agree your version looks nicer.